concepts/translations/built-in-translation-system.md

```diff
@@ -0,0 +1,229 @@
+**Example:**
+
+```yaml
+plugins: [
+  'PluginPublisher',
+  'SwagB2bPlatform',
+  'SwagCmsExtensions'
+]
+```
+
+**Description:**  
+A list of supported plugins for which translations are available.
+
+---
+
+#### `excluded-locales`
+
+**Type:** `array[string]`  
+**Example:**
+
+```yaml
+excluded-locales: [ 'de-DE', 'en-GB' ]
+```
+
+**Description:**  
+A list of language locales to be excluded from translation processing. German (Germany) and English (UK) are excluded by
+default since they are shipped with Shopware. The exclusion applies to plugins too.
+
+---
+
+#### `plugin-mapping`
+
+**Type:** `array[object]`  
+**Example:**
+
+```yaml
+plugin-mapping:
+  - plugin: 'SwagPublisher'
+    name: 'PluginPublisher'
+```
+
+**Fields:**
+
+- `plugin` (`string`): The internal plugin identifier (e.g., directory or code name).
+- `name` (`string`): The corresponding plugin name in the translation repository.
+
+**Description:**  
+Allows mapping between internal plugin identifiers and repository names in case they differ.
+
+---
+
+#### `languages`
+
+**Type:** `array[object]`  
+**Example:**
+
+```yaml
+languages:
+  - name: 'Français'
+    locale: 'fr-FR'
+```
+
+**Fields:**
+
+- `name` (`string`): Human-readable language name (preferably in the native script).
+- `locale` (`string`): Language code according to [IETF BCP 47](https://datatracker.ietf.org/doc/html/bcp47), used by
+  Shopware for translations.
+
+**Description:**  
+Defines the set of supported languages for which translations should be retrieved. Each entry specifies a display name
+and a locale code.
+
+---
+
+## How to extend or modify the configuration handling
+
+### TranslationConfigLoader
+
+The `TranslationConfigLoader` (`src/Core/System/Snippet/Service/TranslationConfigLoader.php`) is part of the Shopware
+core and is responsible for loading and validating the `translation.yaml` file. It provides a `TranslationConfig` object
+that contains all configured fields from the `translation.yaml`. It ensures that URLs are valid, languages and plugins
+are properly structured, and plugin mappings are resolved. Errors such as missing files or invalid configuration values
+are raised as `SnippetException`.
+To extend or modify its behavior, the decoration pattern is used: services should depend on the abstract class
+`AbstractTranslationConfigLoader`, and custom decorators can override methods like `load()` or the configuration path
+while delegating to the original loader.
+
+### TranslationConfig
+
+The `TranslationConfig` (`src/Core/System/Snippet/Service/TranslationConfig.php`) is a data structure that holds the
+configuration details loaded from the `translation.yaml` file.
+You can require it via dependency injection and because of the usage of the `TranslationConfigLoader` with lazy loading,
+the configuration is always available when needed.
+
+[migration-guide]: ../../resources/references/upgrades/core/translation/language-pack-migration.md
+[language-layer-docs]: TODO
```

Comment:
Here is still a todo but i guess this will be handled by @marcelbrode

Reply:
Yup, I'll do this after my command MR